README.md

```diff
@@ -0,0 +1,318 @@
+| `AWS_ROOT_VOLUME_SIZE` | AWS root volume size (GB) | 120 |
+| `ENVIRONMENT_TAG` | Environment tag for resources | production |
+| `MANAGED_BY_TAG` | Managed-by tag for resources | terraform |
+
+## Platform-Specific Notes
+
+### AWS
+
+- Credentials automatically extracted from cluster secrets
+- Supports custom AMI selection
+- Configurable instance types and volume sizes
+
+### Azure
+
+- **Instance names limited to 13 characters** (automatically truncated)
+- Supports specific image versions or 'latest'
+- Resource groups and prefixes from cluster configuration
+
+### GCP
+
+- Service account credentials from cluster secrets
+- Zone and region auto-detected
+- Supports custom machine types
+
+### vSphere
+
+- Requires pre-configured Windows templates
+- vCenter credentials from cluster secrets
+- Template names: `Windows-Server-2019-Template`, `Windows-Server-2022-Template`
+
+### Nutanix
+
+- Requires pre-configured Windows images in Prism Central
+- Cluster and subnet UUIDs auto-detected
+- Image names: `Windows-Server-2019`, `Windows-Server-2022`
+
+### Bare Metal (None)
+
+- Uses AWS credentials from `~/.aws/config` and `~/.aws/credentials`
+- Platform detected as "none"
+- Provisions using AWS infrastructure
+
+## Architecture
+
+This project uses a modular architecture with separated concerns:
+
+```
+terraform-windows-provisioner/
+├── byoh.sh                 # Main entry point
+├── lib/                    # Library modules
+│   ├── config.sh          # Configuration loading
+│   ├── credentials.sh     # Credential management
+│   ├── platform.sh        # Platform detection & config
+│   ├── terraform.sh       # Terraform operations
+│   └── validation.sh      # Input validation
+├── configs/               # Configuration files
+│   ├── defaults.conf      # Default values
+│   └── examples/          # Platform-specific examples
+└── platforms/             # Platform-specific Terraform
+    ├── aws/
+    ├── azure/
+    ├── gcp/
+    ├── vsphere/
+    ├── nutanix/
+    └── none/
+```
+
+## Troubleshooting
+
+### Check Cluster Credentials
+
+```bash
+oc get secret -n kube-system
+oc get secret -n openshift-machine-api
+```
+
+### Verify Cluster Status
+
+```bash
+oc get clusterversion
+oc get nodes
+```
+
+### View Terraform State
+
+```bash
+cd /tmp/terraform_byoh/<platform>
+terraform show
+terraform output
+```
+
+### Check ConfigMap
+
+```bash
+# Find WMCO namespace
+oc get deployment --all-namespaces | grep windows-machine-config-operator
+
+# View ConfigMap
+oc get configmap windows-instances -n <wmco-namespace> -o yaml
+```
+
+### Enable Debug Logging
+
+```bash
+export BYOH_LOG_LEVEL=DEBUG
+./byoh.sh apply myapp 2
+```
+
+## Contributing
+
+We welcome contributions! Please see [CONTRIBUTING.md](CONTRIBUTING.md) for details on:
+
+- Code of conduct
+- Development setup
+- Pull request process
+- Testing requirements
+
+## Security
+
+For security issues, please see [SECURITY.md](SECURITY.md).
```

Comment:
This file, SECURITY.md, isn't present that I can see; as such the link doesn't work.